extension/modules/personas.js

```diff
@@ -0,0 +1,1480 @@
+                return this._header = document.getElementById("messengerWindow");
+            case PersonaService.FIREFOX_ID:
+                return this._header = document.getElementById("main-window");
+            default:
+                throw "unknown application ID " + PersonaService.appInfo.ID;
+        }
+    },
+    get_header(document) {		
+        switch (PersonaService.appInfo.ID) {
+            case PersonaService.THUNDERBIRD_ID:
+                return document.getElementById("messengerWindow");
+            case PersonaService.FIREFOX_ID:
+                return document.getElementById("main-window");
+            default:
+                throw "unknown application ID " + PersonaService.appInfo.ID;
+        }
+    },
+
+    get _footer() {		
+        delete this._footer;
+        switch (PersonaService.appInfo.ID) {
+            case PersonaService.THUNDERBIRD_ID:
+                return this._footer = document.getElementById("status-bar");
+            case PersonaService.FIREFOX_ID:
+                return this._footer = document.getElementById("browser-bottombox");
+            default:
+                throw "unknown application ID " + PersonaService.appInfo.ID;
+        }
+    },
+    get_footer() {		
+        switch (PersonaService.appInfo.ID) {
+            case PersonaService.THUNDERBIRD_ID:
+                return document.getElementById("status-bar");
+            case PersonaService.FIREFOX_ID:
+                return document.getElementById("browser-bottombox");
+            default:
+                throw "unknown application ID " + PersonaService.appInfo.ID;
+        }
+    },
+
+    get _thunderbirdRegExp() {
+        delete this._thunderbirdRegExp;
+        return this._thunderbirdRegExp = new RegExp("^" + this._siteURL.replace(/\./g, "\\."));
+    },
+
+    get _siteURL() {
+        return "https://" + this._prefs.get("addons-host") + "/";
+    },
+
+    get _previewTimeout() {
+        return this._prefs.get("previewTimeout");
+    },
+
+    // XXX We used to use this to direct users to locale-specific directories
+    // on the personas server, but we're not using it anymore, as we no longer
+    // have locale-specific pages on the server.  And once we get them back,
+    // it'll probably make more sense for the browser and server to do locale
+    // negotiation using the standard mechanisms anyway, so this is no longer
+    // needed.
+    get _locale() {
+        switch (this.Preferences.get("general.useragent.locale", "en-US")) {
+            case 'ja':
+            case 'ja-JP-mac':
+                return "ja";
+        }
+        return "en-US";
+    },
+
+    /**
+     * Escape CSS special characters in unquoted URLs,
+     * per http://www.w3.org/TR/CSS21/syndata.html#uri.
+     */
+    _escapeURLForCSS: function(url) url.replace(/[(),\s'"]/g, "\$&"),
+
+    openURLInTab: function(url,event) {
+		var document=event.currentTarget.ownerDocument;
+		var window=document.defaultView;
```

Comment:
`extension/modules/personas.js` is loaded into `bootstrap.js` context. `persona.js` is not injected into windows. So there is no `window` or `document` conflict over here.

We detect the `window` and `document` by the help of `event` that take place.